dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TaskInstanceServiceImpl.java

```diff
@@ -182,19 +182,19 @@ public Result queryTaskListPaging(User loginUser,
     /**
      * change one task instance's state from failure to forced success
      *
-     * @param loginUser login user
-     * @param projectCode project code
+     * @param loginUser      login user
+     * @param projectCode    project code
      * @param taskInstanceId task instance id
      * @return the result code and msg
      */
     @Transactional
     @Override
-    public Map<String, Object> forceTaskSuccess(User loginUser, long projectCode, Integer taskInstanceId) {
+    public Result forceTaskSuccess(User loginUser, long projectCode, Integer taskInstanceId) {
+        Result result = new Result();
         Project project = projectMapper.queryByCode(projectCode);
         // check user access for project
-        Map<String, Object> result =
-                projectService.checkProjectAndAuth(loginUser, project, projectCode, FORCED_SUCCESS);
-        if (result.get(Constants.STATUS) != Status.SUCCESS) {
+        projectService.checkProjectAndAuth(result, loginUser, project, projectCode, FORCED_SUCCESS);
```

Comment:
Don't to use the new `checkProjectAndAuth` function?

Reply:
I found they are these two function is the same except that result handles it differently, and I think the return result one is better, because the input parameter cannot be modified directly in general.

WDYT? @SbloodyS

Reply:
> I found they are these two function is the same except that result handles it differently, and I think the return result one is better, because the input parameter cannot be modified directly in general.
> 
> WDYT? @SbloodyS

Agreed.